modules/core/shared/src/main/scala/retry/package.scala

```diff
@@ -8,6 +8,33 @@ import scala.concurrent.duration.FiniteDuration
 package object retry {
   def retryingM[A] = new RetryingPartiallyApplied[A]
```

Comment:
Aside (idea for a separate PR): I've never really liked this naming. It was inherited without much thought from the `retry` Haskell package. I think it would be good to deprecate this method and replace it with one that fits the naming scheme introduced in this PR, to make the API more consistent. `retryingOnFailures` I suppose.

Reply:
I am not a big fan of the name either lol, but I am pretty terrible at naming things.

Internally, we treat these as `retrying` (A => Boolean) and `recovering` (E => Boolean). It works OK but its not perfect.

Reply:
I created a PR for this: https://github.com/cb372/cats-retry/pull/261